gmusicapi/clients/mobileclient.py

```diff
@@ -123,7 +130,7 @@ def login(self, email, password, android_id, locale='en_US'):
             self.logger.info("failed to authenticate")
             return False
 
-        self.android_id = android_id
+        self.android_id = self._validate_device_id(android_id)
```

Comment:
Yeah I wanted to move it up too but ran into that issue. That's a good idea, I'll update it to log out before raising the error.